src/WebJobs.Extensions.DocumentDB/DocumentDBUtility.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DocumentDB
+{
+    internal static class DocumentDBUtility
+    {
+        public static async Task ExecuteAndIgnoreStatusCodeAsync(HttpStatusCode statusToIgnore, Func<Task> function)
+        {
+            try
+            {
+                await function();
+            }
+            catch (DocumentClientException ex)
+            {
+                if (ex.StatusCode != statusToIgnore)
+                {
+                    throw;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns the core DocumentDB type for the supplied parameter.
+        /// </summary>
+        /// <remarks>
+        /// For example, the core Type is T in the following parameters:
+        /// <list type="bullet">
+        /// <item><description><see cref="ICollector{T}"/></description></item>
+        /// <item><description>T[]</description></item>
+        /// <item><description>out T</description></item>
+        /// <item><description>out T[]</description></item>
+        /// </list>
+        /// </remarks>
+        /// <param name="type">The Type to evaluate.</param>
+        /// <returns>The core Type</returns>
+        public static Type GetCoreType(Type type)
```

Comment:
I've done this but it required an extern alias in the test project. I'll file an issue to undo that once this common code can be removed.

It's a byproduct of us having a single test project for simplicity. Since two projects that the tests reference share the same file, there's a conflict.

---
In reply to: [56360306](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/47#discussion_r56360306) [](ancestors = 56360306,56358369)